src/werkzeug/wsgi.py

```diff
@@ -1,19 +1,223 @@
 import io
+import mimetypes
+import os
 import re
+import unicodedata
 from functools import partial
 from functools import update_wrapper
 from itertools import chain
+from time import time
+from zlib import adler32
 
 from ._internal import _encode_idna
 from ._internal import _make_encode_wrapper
 from ._internal import _to_bytes
 from ._internal import _to_str
+from .datastructures import Headers
+from .exceptions import RequestedRangeNotSatisfiable
 from .urls import uri_to_iri
 from .urls import url_join
 from .urls import url_parse
 from .urls import url_quote
 
 
+def send_file(
+    filename_or_fp,
+    root_path,
+    environ=None,
```

Comment:
Not too sure about this approach -- `send_file` in Flask was using the app's `root_path` but we don't have that equivalent in werkzeug?

Reply:
Passing `environ` makes sense. See other review comments about `root_path`.